### ISSUE TITLE
Game crashes when loading moon with the latest SDK version

### ISSUE THREAD

**Issue by Wesley-wasnietavailabledusdanmaarzo** [HolographicWings/LethalSDK-Unity-Project#72]
So I've updated a moon to the latest version of the SDK.
But whenever I attempt to land on this moon, the game crashes. (this doesn't happen with leCore)

I also noticed that even though I am 10000% sure I've updated it to the latest version, it still tells me "the SDK is not up to date: 1.3.36".

**Comment by HolographicWings** [comment]
True, i forgot to increment the version number inside of the SDK

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
I knew I wasn't crazy lol

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
I have noticed another problem though.
![image](https://github.com/HolographicWings/LethalSDK-Unity-Project/assets/45416129/ef8c0523-a4ad-40ae-89f5-d3780fecfd54)
It seems like the moons aren't able to find the assets from Lethal Expansion.

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
To summarise, any moon that updates to sdk 1.3.34+ will be broken with the latest version of Lethal Expansion.
Though newly created moons will work fine.

**Comment by HolographicWings** [comment]
> Actually it might just be every moon that's been made before SDK version 1.3.34. Cause even when updated to the latest version, they still have the same issue.

do Wateridge have the issue too ?
https://thunderstore.io/c/lethal-company/p/HolographicWings/LEDefaultModules/

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
I'll take a look real quick
I know it's on any weather

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
uh, wateridge isn't there anymore?
Only using LE for this test

**Comment by HolographicWings** [comment]
> uh, wateridge isn't there anymore? Only using LE for this test

i send you the link for wateridge

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
Where?

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
omg I am blind

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
Hmm, wateridge seems fine
I tried a few other moons as well like Orion and they had similar issues

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
Yeah but that's exactly the thing, I have updated my maps to the latest SDK version
But for whatever reason it's still a thing. And it's not only the sun texture, It's also unable to find the droppod for instance.

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
Has wateridge been updated to use SDK version 1.3.36?

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
From the research I've done, I've concluded that a bunch of names for materials, scripts etc. from le were changed between updates which caused the issues

**Comment by HolographicWings** [comment]
> From the research I've done, I've concluded that a bunch of names for materials, scripts etc. from le were changed between updates which caused the issues

I will upload a new SDK version, this is weird

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
> > From the research I've done, I've concluded that a bunch of names for materials, scripts etc. from le were changed between updates which caused the issues
> 
> I will upload a new SDK version today, this is weird

This might be related but another thing I've noticed is that EntranceTeleportInstancer prefabs reset when a moon is loaded in game
For instance if I add a material to the "door" object or change anything about them, in game they'll be set back to the default prefab from LE
And when loaded with LECore for instance, the doors dissapear entirely as if the instancer prefab doesn't exist
So there's definitely something weird going on

**Comment by HolographicWings** [comment]
> > > From the research I've done, I've concluded that a bunch of names for materials, scripts etc. from le were changed between updates which caused the issues
> > 
> > 
> > I will upload a new SDK version today, this is weird
> 
> This might be related but another thing I've noticed is that EntranceTeleportInstancer prefabs reset when a moon is loaded in game
> For instance if I add a material to the "door" object or change anything about them, in game they'll be set back to the default prefab from LE

because the prefabs are in the LethalExpansion.lem, you can't and are not supposed to edit what is inside of this module
create a copy of the entrance prefab inside of your mod folder, and register it in the asset bank as a network prefab
you can watch how it's made in the Christmas Village

> And when loaded with LECore for instance, the doors dissapear entirely as if the instancer prefab doesn't exist
So there's definitely something weird going on

LECore is not my mod, i can't help you with issues about it, in fact, LECore is not up to date with the new LE additions.

> From the research I've done, I've concluded that a bunch of names for materials, scripts etc. from le were changed between updates which caused the issues

After investigation, mappers must check on their side if the material is found, or use the original Lighting prefab from the LethalExpansion module of the SDK to automatically get updated. Unity is tricky in the way of manage asset IDs and i barely have a control on it.

**Comment by Wesley-wasnietavailabledusdanmaarzo** [comment]
> because the prefabs are in the LethalExpansion.lem, you can't and are not supposed to edit what is inside of this module create a copy of the entrance prefab inside of your mod folder, and register it in the asset bank as a network prefab you can watch how it's made in the Christmas Village

Ah that makes sense, I've been doing the way I described, which used to work strangely enough

> LECore is not my mod, i can't help you with issues about it, in fact, LECore is not up to date with the new LE additions.

That's true

> After investigation i can't resolve the issue with an SDK update, mappers must check on their side if the material is found, or use the original Lighting prefab from the LethalExpansion module of the SDK to automatically get updated. Unity is tricky in the way of manage asset IDs and i barely have a control on it.

I found that it's an easy fix on the mappers side but it's still a bit strange how most of the names are capitalised now
Thanks for looking into it though

**Comment by ryudious** [comment]
Have latest version, still Unity warns me everytime I hit play lol
